modules/core/src/main/scala/org/tessellation/domain/snapshot/programs/Download.scala

```diff
@@ -225,6 +220,19 @@ object Download {
               case Some(snapshot) =>
                 globalSnapshotContextFns
                   .createContext(context, lastSnapshot, snapshot)
+                  .flatTap(newContext =>
+                    snapshotStorage
+                      .hasCorrectSnapshotInfo(snapshot.ordinal, snapshot.stateProof)
+                      .flatMap(Applicative[F].unlessA(_) {
+                        newContext.stateProof.flatMap { contextProof =>
+                          if (contextProof === snapshot.stateProof) {
+                            snapshotStorage.persistSnapshotInfo(snapshot.ordinal, newContext)
```

Comment:
Won't it always fails in such a scenario? I mean the snasphot info exists at such ordinal, so overwriting snapshot info at the same ordinal will throw an error.

Reply:
Overwriting snapshot info does not throw an error and just before is a check whether the snapshot info stored on disk is correct (so we can reuse it) or incorrect or non existing (and we should (over)write it)